boards/arduino-atmega-common/include/periph_conf.h

```diff
@@ -139,6 +139,14 @@ extern "C" {
 #endif
 /** @} */
 
+/**
+ * @name    I2C configuration
+  * @{
+ */
+#define I2C_NUMOF           1
+#define I2C_0_EN            1
```

Comment:
no need for this define: your implementation does support only a single I2C device anyway, so this switch is obsolete.